### PR TITLE
Set kotlin dev version to 1.6.20-M1

### LINKED ISSUE #1436: Support context receivers
## Expected Behavior
Ktlint should handle Kotlin files where context receivers are used.

## Current Behavior
Scanning a file which includes a function using context receivers from Kotlin 1.6.20 currently fails with 
```
Not a valid Kotlin file (5:1 expecting a top level declaration) (cannot be auto-corrected)
```

Example declaration from [Kotlin blog](https://blog.jetbrains.com/kotlin/2022/02/kotlin-1-6-20-m1-released/)
```
interface LoggingContext {
    val log: Logger // This context provides a reference to a logger 
}

context(LoggingContext)
fun startBusinessOperation() {
    // You can access the log property since LoggingContext is an implicit receiver
    log.info("Operation has started")
}

fun test(loggingContext: LoggingContext) {
    with(loggingContext) {
        // You need to have LoggingContext in a scope as an implicit receiver
        // to call startBusinessOperation()
        startBusinessOperation()
    }
}
```

## Additional information
* Current version of ktlint: 0.45.1

### PR DESCRIPTION
**Pull request by paul-dingemans**
## Description

Set Kotlin dev version to `1.6.20-M1`

Closes #1436

## Checklist

- [X] PR description added
- [ ] tests are added
- [X] `CHANGELOG.md` is updated

In case of adding a new rule:
- [ ] `README.md` is updated
- [ ] Rule has been applied on Ktlint itself and violations are fixed

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -15,7 +15,7 @@ repositories {
 // Pass '-PkotlinDev' to command line to enable kotlin-in-development version
 val kotlinVersion = if (project.hasProperty("kotlinDev")) {
     logger.warn("Enabling kotlin dev version!")
-    "1.6.0"
+    "1.6.20-M1"
```
Comment (Goooler) [inline buildSrc/build.gradle.kts:18]: May you can bump it to 1.6.20-RC2?
https://github.com/JetBrains/kotlin/releases/tag/v1.6.20-RC2

Comment (romtsn) [inline buildSrc/build.gradle.kts:18]:
```suggestion
    "1.6.20-RC2"
```

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -15,7 +15,7 @@ repositories {
...
 } else {
     "1.6.0"
```
Comment (romtsn) [inline buildSrc/build.gradle.kts:20]:
```suggestion
    "1.6.10"
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 ### Fixed
 
 ### Changed
+* Set Kotlin development version to `1.6.20-M1`.
```
Comment (romtsn) [inline CHANGELOG.md:12]:
```suggestion
* Set Kotlin development version to `1.6.20-RC2` and Kotlin version to `1.6.10`.
```

**Pull request merged**